File: String-Algorithms/String-Algorithms-master/Rotate Image/rotateImage.cpp
# include<iostream>
# include<cstdlib>
# include<string>
# include<map>

using namespace std;

class solution
{
public:
    void rotate_image(int** img_arr,int n)
    {
        for(int layer = 0;layer < n/2;layer++)
        {
            int first = layer;
            int last = n - 1 - layer;
            for(int i=first;i<last;i++)
            {
                int offset = i - first;
                int top = img_arr[first][i];
                img_arr[first][i] = img_arr[last - offset][first];
                img_arr[last - offset][first] = img_arr[last][last - offset];
                img_arr[last][last - offset] = img_arr[i][last];
                img_arr[i][last] = top;
            }
        }
    }
};

int main()
{
    solution sol;
    int** img_arr = new int*[n];
    for(int i=0;i<n;i++)
    {
        img_arr[i] = new int[n];
        for(int j=0;j<n;j++)
        {
            cin >> img_arr[i][j];
        }
    }
    sol.rotate_image(img_arr,n);
    for(int i=0;i<n;i++)
    {
        for(int j = 0;j<n;j++)
        {
            cout << img_arr[i][j] << " ";
        }
        cout << endl;
    }
    return 0;
}

File: String-Algorithms/String-Algorithms-master/Rotate Image/rotateImage.py
class solution:
    def rotateImage(self,img_arr,n):
        for layer in range(int(n/2)):
            first = layer
            last = n-1-layer
            for i in range(first,last):
                offset = i - first
                top = img_arr[first][i]
                img_arr[first][i] = img_arr[last - offset][first]
                img_arr[last - offset][first] = img_arr[last][last - offset]
                img_arr[last][last - offset] = img_arr[i][last]
                img_arr[i][last] = top

n = int(input())
matrix = []
for i in range(n):          # A for loop for row entries
    a =[]
    for j in range(n):      # A for loop for column entries
         a.append(int(input()))
    matrix.append(a)

sol = solution()
sol.rotateImage(matrix,n)
for i in range(n):
    for j in range(n):
        print(matrix[i][j], end = " ")
    print()
